### DISCUSSION TITLE
time.sleep()

### DISCUSSION THREAD

**Discussion by Gal-Peled** [jupyterlite/jupyterlite#831] [Q&A]
I cant get time.sleep() function to work , is these a known issue? 

here is a simple code that runs on JupyterLab but not on Juyterlite . it just print the numbers with no delay.  
jupyterlite version: '0.1.0b13'
my operating system - window 10.0.19043 

```
import time
number = 10
for count in range(1, number+1):
    if count < number:
        print(count, end=',')
        time.sleep(1)
    else:
        print('END')
```

**Comment by bollwyvl** [comment]
Yep, not everything works the way one would expect due to limitations of the WASM environment.

You could probably use:

```py
import asyncio
...
        await asyncio.sleep(1)
```

**Comment by jtpio** [comment]
There is also the xeus python kernel with support for `time.sleep()`: https://xeus-python-kernel.readthedocs.io/en/latest/

**Reply by Gal-Peled** [reply]
Thx will try both 
what is the differance between pip installing Jupyterlite vs. installing JupyterLab? To me the main advantague of Jupyterliter to use it without installing anything on my Local machine
